core/src/commonMain/kotlin/com/ustadmobile/core/viewmodel/report/ReportEditViewModel.kt

```diff
@@ -0,0 +1,230 @@
+                    if (series.reportSeriesUid == seriesId) {
+                        val updatedFilters =
+                            series.reportSeriesFilters?.toMutableList() ?: mutableListOf()
+                        updatedFilters.add(filter2)
+                        series.copy(reportSeriesFilters = updatedFilters)
+                    } else {
+                        series
+                    }
+                }
+
+                prev.copy(
+                    reportOptions2 = prev.reportOptions2.copy(
+                        series = updatedSeriesList
+                    )
+                )
+            }
+        }
+    }
+
+
+    fun onAddFilter(seriesId: Int) {
+        navigateForResult(
+            nextViewName = ReportFilterEditViewModel.DEST_NAME,
+            key = RESULT_KEY_REPORT,
+            currentValue = null,
+            serializer = Report.serializer(),
+            args = mapOf("reportSeriesUid" to seriesId.toString())
+        )
+    }
+
+
+    fun onAddSeries() {
+        _uiState.update { prev ->
+            prev.copy(
+                reportOptions2 = prev.reportOptions2.copy(
+                    series = prev.reportOptions2.series + ReportSeries2(
+                        reportSeriesUid = prev.reportOptions2.series.size + 1,
+                        reportSeriesVisualType = null,
+                        reportSeriesSubGroup = null,
+                        reportTimeRange = null,
+                        reportSeriesYAxis = null
+                    ),
+                )
+            )
+        }
+    }
+    fun onRemoveSeries(seriesId: Int) {
+        _uiState.update { prev ->
+            val updatedSeriesList = prev.reportOptions2.series.filterNot { it.reportSeriesUid == seriesId }
+
+            prev.copy(
+                reportOptions2 = prev.reportOptions2.copy(
+                    series = updatedSeriesList
+                )
+            )
+        }
+    }
+
+    fun onRemoveFilter(index: Int, seriesId: Int) {
+        _uiState.update { prev ->
+            val updatedSeriesList = prev.reportOptions2.series.map { series ->
+                if (series.reportSeriesUid == seriesId) {
+                    val updatedFilters = series.reportSeriesFilters?.toMutableList()?.apply {
+                        removeAt(index)
+                    }
+                    series.copy(reportSeriesFilters = updatedFilters)
+                } else {
+                    series
+                }
+            }
+
+            prev.copy(
+                reportOptions2 = prev.reportOptions2.copy(
+                    series = updatedSeriesList
+                )
+            )
+        }
+    }
+
+
+    companion object {
+        const val DEST_NAME = "Report"
+        const val DEST_NAME_HOME = "ReportHome"
+        val ALL_DEST_NAMES = listOf(DEST_NAME, DEST_NAME_HOME)
+        const val RESULT_KEY_REPORT = "arg"
+        const val ENTITY_UID = "uid"
+    }
+}
+
+data class ReportEditUiState(
```

Comment:
Please put ReportEditUiState at the top like as is done with all other view models.